### PR TITLE
Updated Iden3 packages to v2

### PR DESCRIPTION
**Pull request by lukachi**

### CODE REVIEW

File: packages/snap/src/identity.ts [hunk 1]
```diff
@@ -50,19 +50,13 @@ export class Identity {
...
-    return this.did.toString();
-  }
-
-  public get identityId() {
-    return this.did.id;
-  }
-
-  public get identityIdString() {
-    return this.identityId.string();
+    return this.did.string();
```
Comment (ihordiachenko) [inline packages/snap/src/identity.ts:53]: What format will it have: `did:iden3:..` or `did:iden3:readonly:...`?

File: packages/snap/package.json [hunk 2]
```diff
@@ -50,11 +50,11 @@
     "@ethersproject/keccak256": "5.7.0",
     "@ethersproject/providers": "5.7.2",
     "@glazed/did-datastore": "0.3.2",
-    "@iden3/js-crypto": "1.0.0-beta.1",
-    "@iden3/js-iden3-core": "1.0.0-beta.2",
-    "@iden3/js-jsonld-merklization": "1.0.0-beta.14",
-    "@iden3/js-jwz": "1.0.0-beta.2",
-    "@iden3/js-merkletree": "1.0.0-beta.1",
+    "@iden3/js-crypto": "1.0.3",
```
Comment (ihordiachenko) [inline packages/snap/package.json:53]: PR title says core V2 but the libs are V1. Is it ok?

File: packages/snap/src/identity.ts (hunk 1 above)
Comment (lukachi) [reply packages/snap/src/identity.ts:53]: second one

File: packages/snap/src/index.ts [hunk 3]
```diff
@@ -190,7 +199,16 @@ export const onRpcRequest = async ({
           createProofRequest.query,
           issuerDid,
         )
-      ).filter((cred) => cred.credentialSubject.id === identityStorage.did);
+      ).filter((cred) => {
+        const splittedCredSubjId = String(cred.credentialSubject.id).split(':');
```
Comment (ihordiachenko) [inline packages/snap/src/index.ts]: `split(':')` is spread all over the place. Consider encapsulating this logic into a function like `parseDidV2(...)`

**Review by ihordiachenko — APPROVED** [review]

**Pull request merged**